src/popup/components/form/FormAccountInput.vue

```diff
@@ -0,0 +1,265 @@
+  props: {
+    type: { type: String, default: '' },
+    modelValue: { type: String, default: '' },
+    placeholder: { type: String, default: '' },
+    textLimit: { type: Number, default: null },
+    resizable: { type: Boolean, default: true },
+    size: {
+      type: String,
+      default: null,
+      validator: (val: string) => SIZES.includes(val),
+    },
+    options: {
+      type: Array,
+      default: () => [],
+    },
+    enterSubmit: Boolean,
+    readonly: Boolean,
+    hideClearIcon: Boolean,
+    protocol: { type: String as PropType<Protocol>, default: null },
+  },
+  emits: ['update:modelValue', 'submit'],
+  setup(props, { emit }) {
+    const {
+      addressBookFiltered,
+      protocolFilter,
+    } = useAddressBook();
+    protocolFilter.value = props.protocol;
+
+    const isDropdownOpen = ref(false);
+    const selectedAddress = ref<IAddressBookEntry | undefined>();
+    const filteredOptions = computed(() => (
+      addressBookFiltered.value.filter((entry: IAddressBookEntry) => (
+        entry.address.toLowerCase().includes(props.modelValue.toString().toLowerCase())
+            || entry.name.toLowerCase().includes(props.modelValue.toString().toLowerCase())
+      ))
+    ));
+
+    function clearAddress() {
+      selectedAddress.value = undefined;
+      emit('update:modelValue', '');
+    }
+
+    function selectValue(entry: IAddressBookEntry, $event: Event) {
+      $event.preventDefault();
+      emit('update:modelValue', entry?.address);
+      isDropdownOpen.value = false;
+    }
+
+    onMounted(() => {
+      watch(() => props.modelValue, () => {
+        const searchAddress = addressBookFiltered.value
+          ?.find((entry: IAddressBookEntry) => (entry.address === props.modelValue));
+        if (searchAddress) {
+          selectedAddress.value = searchAddress;
+        }
+        if (!props.modelValue) { // If it's cleared externally
+          selectedAddress.value = undefined;
+        }
+
+        isDropdownOpen.value = props.modelValue.toString().length >= 2
+            && filteredOptions.value.length > 0;
+        if (addressBookFiltered.value?.find(
+          (entry) => (entry.address === props.modelValue.toString()),
+        )) {
+          isDropdownOpen.value = false;
+        }
+      }, { immediate: true });
+    });
+
+    return {
+      filteredOptions,
+      selectedAddress,
+      clearAddress,
+      selectValue,
+      isDropdownOpen,
+      CircleCloseIcon,
+    };
+  },
+});
+</script>
+
+<style lang="scss" scoped>
+@use '@/styles/variables' as *;
+@use '@/styles/typography';
+
+.close-icon {
+  padding: 0;
+  opacity: 0.5;
+}
+
+.truncated-text {
+  color: $color-white;
+}
+
+.input-field {
+  &:where(
+    &:has(input:focus),
+    &:has(textarea:focus),
+    &:has(label:active)
+  ) {
+    .dropdown-wrapper {
+      display: block;
+      top: calc(100% + 4px);
+      opacity: 1;
+      user-select: all;
+      pointer-events: all;
+    }
+  }
+
+  .dropdown-wrapper {
```

Comment:
I suggest to order CSS declarations according to 9elements rules: https://9elements.com/css-rule-order/